frontend/src/app/AdminNavBar/AdminNavBar.tsx

```diff
@@ -80,11 +85,47 @@ export interface AdminNavBarProps {
 export const AdminNavBar = ({ isMenuOpen }: AdminNavBarProps): JSX.Element => {
   const { user } = useUser()
 
+  const ROLLOUT_ANNOUNCEMENT_KEY = useMemo(
+    () => ROLLOUT_ANNOUNCEMENT_KEY_PREFIX + user?._id,
+    [user],
+  )
+  const [hasSeenAnnouncement] = useLocalStorage<boolean>(
+    ROLLOUT_ANNOUNCEMENT_KEY,
+  )
+
+  // Only want to show the modal if user id exists but user has no emergency contact
+  const shouldShowModal = user && user._id && !user.contact
+
+  const emergencyContactKey = shouldShowModal
+    ? EMERGENCY_CONTACT_KEY_PREFIX + user._id
+    : null
+
+  const [hasSeenContactModal, setHasSeenContactModal] =
+    useLocalStorage<boolean>(emergencyContactKey)
+
   const {
     isOpen: isContactModalOpen,
     onClose: onContactModalClose,
     onOpen: onContactModalOpen,
-  } = useDisclosure()
+  } = useDisclosure({
+    onClose: () => {
+      setHasSeenContactModal(true)
+    },
+  })
+
+  // Emergency contact modal appears after the rollout announcement modal
+  useEffect(() => {
+    if (!hasSeenContactModal && !user?.contact && hasSeenAnnouncement) {
```

Comment:
the conditional here may result in modal opening when user is still loading. should we tighten the conditional?

```suggestion
    if (!hasSeenContactModal && user && !user.contact && hasSeenAnnouncement) {
```